lib/util/http/index.spec.ts

```diff
@@ -516,34 +516,68 @@ describe('util/http/index', () => {
   });
 
   describe('Throttling', () => {
+    const delta = 100;
+
+    beforeEach(() => {
+      jest.useFakeTimers({ advanceTimers: true });
+    });
+
     afterEach(() => {
       jest.useRealTimers();
     });
 
     it('works without throttling', async () => {
-      jest.useFakeTimers({ advanceTimers: 1 });
-      httpMock.scope(baseUrl).get('/foo').twice().reply(200, 'bar');
+      httpMock.scope(baseUrl).get('/foo').times(10).reply(200, 'bar');
 
       const t1 = Date.now();
-      await http.get('http://renovate.com/foo');
-      await http.get('http://renovate.com/foo');
+      await Promise.all([
+        http.get('http://renovate.com/foo'),
+        http.get('http://renovate.com/foo'),
+        http.get('http://renovate.com/foo'),
+        http.get('http://renovate.com/foo'),
+        http.get('http://renovate.com/foo'),
+        http.get('http://renovate.com/foo'),
+        http.get('http://renovate.com/foo'),
+        http.get('http://renovate.com/foo'),
+        http.get('http://renovate.com/foo'),
+        http.get('http://renovate.com/foo'),
+      ]);
       const t2 = Date.now();
 
-      expect(t2 - t1).toBeLessThan(100);
+      expect(t2 - t1).toBeLessThan(delta);
     });
 
     it('limits request rate by host', async () => {
-      jest.useFakeTimers({ advanceTimers: true });
-      httpMock.scope(baseUrl).get('/foo').twice().reply(200, 'bar');
-      hostRules.add({ matchHost: 'renovate.com', maxRequestsPerSecond: 0.25 });
+      httpMock.scope(baseUrl).get('/foo').times(4).reply(200, 'bar');
+      hostRules.add({ matchHost: 'renovate.com', maxRequestsPerSecond: 3 });
+
+      const t1 = Date.now();
+      await Promise.all([
+        http.get('http://renovate.com/foo'),
+        http.get('http://renovate.com/foo'),
+        http.get('http://renovate.com/foo'),
+        http.get('http://renovate.com/foo'),
+      ]);
+      const t2 = Date.now();
+
+      expect(t2 - t1).toBeWithin(1000, 1000 + delta);
+    });
+
+    it('defaults to 5 requests per second', async () => {
+      Http.setDefaultLimits();
+      httpMock.scope(baseUrl).get('/foo').times(5).reply(200, 'bar');
 
       const t1 = Date.now();
-      await http.get('http://renovate.com/foo');
-      jest.advanceTimersByTime(4000);
-      await http.get('http://renovate.com/foo');
+      await Promise.all([
+        http.get('http://renovate.com/foo'),
+        http.get('http://renovate.com/foo'),
+        http.get('http://renovate.com/foo'),
+        http.get('http://renovate.com/foo'),
+        http.get('http://renovate.com/foo'),
+      ]);
       const t2 = Date.now();
 
-      expect(t2 - t1).toBeGreaterThanOrEqual(4000);
+      expect(t2 - t1).toBeWithin(800, 800 + delta);
```

Comment:
Two of these tests will take ~1 sec each, but other alternatives I've tried are bad. Feel free to propose another solution, or just let's get rid of them.